scripts/create_sql_lite_dump.py

```diff
@@ -0,0 +1,125 @@
+    "producedBy_samplingSite_location_ll": "produced_by_sampling_site_location_ll",
+    "producedBy_samplingSite_placeName": "produced_by_sampling_site_place_name",
+    "registrant": "registrant",
+    "samplingPurpose": "sampling_purpose",
+    "curation_label": "curation_label",
+    "curation_description": "curation_description",
+    "curation_accessConstraints": "curation_access_constraints",
+    "curation_location": "curation_location",
+    "curation_responsibility": "curation_responsibility",
+    "relatedResource_isb_core_id": "related_resources_isb_core_id",
+    "source": "source"
+}
+
+NUMERIC_COLUMNS = ["produced_by_sampling_site_elevation_in_meters"]
+
+
+# Filter out any placeholder values that shouldn't appear in the dump
+def _filtered_value(value):
+    if type(value) is str and value == Transformer.NOT_PROVIDED:
+        return ""
+    return value
+
+
+@click.command()
+@click.option(
+    "-d",
+    "--db_url",
+    default=None,
+    help="The SQLite database file URL for storage.  Doesn't need to exist beforehand.",
+)
+@click.option("-s", "--solr_url", default=None, help="Solr index URL")
+@click.option(
+    "-v",
+    "--verbosity",
+    default="DEBUG",
+    help="Specify logging level",
+    show_default=True,
+)
+@click.option(
+    "-H", "--heart_rate", is_flag=True, help="Show heartrate diagnostics on 9999"
+)
+@click.option(
+    "-a",
+    "--authority",
+    default=None,
+    help="Which authority to use when selecting and dumping the resolved_content",
+)
+@click_config_file.configuration_option(config_file_name="isb.cfg")
+@click.pass_context
+def main(ctx, db_url, solr_url, verbosity, heart_rate, authority):
+    isb_lib.core.things_main(ctx, db_url, solr_url, verbosity, heart_rate)
+    engine = create_engine(ctx.obj["db_url"], echo=False)
+    SQLModel.metadata.drop_all(engine, tables=[ISBCoreRecord.__table__])
+    SQLModel.metadata.create_all(engine, tables=[ISBCoreRecord.__table__])
+    session = Session(engine)
+    batch_size = 50000
+    session_commit_frequency = 50000
+    rsession = requests.session()
+    iterator = ISBCoreSolrRecordIterator(rsession, authority, batch_size, 0, "id asc")
+    num_records = 0
+    for solr_record in iterator:
+        new_record = ISBCoreRecord()
+        for key, value in SOLR_TO_SQLITE_FIELD_MAPPINGS.items():
+            solr_value = solr_record.get(key)
+            if type(solr_value) is list:
+                filtered_solr_values = [_filtered_value(value) for value in solr_value]
+                solr_value = "; ".join(filtered_solr_values)
+            elif type(solr_value) is str:
+                # Don't include placeholder empty values in the dump
+                solr_value = _filtered_value(solr_value)
+            # Key is source column solr name, value is dest column sqlite name in sqlite
+            if solr_value is not None:
+                new_record.__setattr__(value, solr_value)
+            else:
```

Comment:
This last bit is a workaround for a hack somewhere below me in the stack (not sure if SQLModel or SQLAlchemy, or SQLite) -- basically if we didn't explicitly initialize a value for a column, it ended up using the column metadata definition as the value and the generated SQL exploded.  I didn't do work to try and pinpoint how/why but verified this workaround works.

Reply:
Make the column nullable? Then it should accept creation without a value

Reply:
Has nothing to do with the column being nullable, unfortunately, I think it's a bug elsewhere in the stack.  Interestingly, the bug persisted even after switching over to the SQLAlchemy Core API.